apps/services/license-api/src/app/modules/license/clients/drivingLicense/drivingLicenseApiClient.service.ts

```diff
@@ -0,0 +1,229 @@
+import { Inject, Injectable } from '@nestjs/common'
+import { GenericLicenseClient, VerifyLicenseResult } from '../../license.types'
+import type { Logger } from '@island.is/logging'
+import { LOGGER_PROVIDER } from '@island.is/logging'
+import { Pass, RevokePassData, Result } from '@island.is/clients/smartsolutions'
+import {
+  DrivingLicenseResponse,
+  LOG_CATEGORY,
+} from './drivingLicenseApiClient.type'
+import { PkPassClient } from './pkpass/pkpass.client'
+import { ConfigType } from '@nestjs/config'
+import { XRoadConfig } from '@island.is/nest/config'
+import { DrivingLicenseApiClientConfig } from './drivingLicenseApiClient.config'
+import { Cache as CacheManager } from 'cache-manager'
+
+@Injectable()
+export class DrivingLicenseApiClientService implements GenericLicenseClient {
+  private readonly xroadApiUrl: string
+  private readonly xroadClientId: string
+  private readonly xroadPath: string
+  private readonly xroadSecret: string
+
+  private pkpassClient: PkPassClient
+
+  constructor(
+    @Inject(LOGGER_PROVIDER) private logger: Logger,
+    @Inject(XRoadConfig.KEY)
+    private xroadConfig: ConfigType<typeof XRoadConfig>,
+    private config: ConfigType<typeof DrivingLicenseApiClientConfig>,
+    private cacheManager?: CacheManager | null,
+  ) {
+    // TODO inject the actual RLS x-road client
+    this.xroadApiUrl = xroadConfig.xRoadBasePath
+    this.xroadClientId = xroadConfig.xRoadClient
+    this.xroadPath = config.xroad.path
+    this.xroadSecret = config.xroad.secret
+
+    this.logger = logger
+    this.cacheManager = cacheManager
+
+    // TODO this should be injected by nest
```

Comment:
```suggestion
```

since we're waiting for the move to Smart Solution v2 this wont happen and can be deleted